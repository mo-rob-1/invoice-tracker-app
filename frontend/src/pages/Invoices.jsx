import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getInvoices, reset } from "../features/invoices/invoiceSlice";
import BackButton from "../components/BackButton";
import InvoiceItem from "../components/InvoiceItem";

function Invoices() {
  const { invoices, isLoading, isSuccess } = useSelector((state) => state.invoice);

  const dispatch = useDispatch();

  useEffect(() => {
    return () => {
      if (isSuccess) {
        dispatch(reset());
      }
    };
  }, [isSuccess, dispatch]);

  useEffect(() => {
    dispatch(getInvoices());
  }, [dispatch]);

  if (isLoading) {
    return <h3>Loading...</h3>;
  }

  return (
    <>
      <BackButton url="/" />
      <h1>Invoices</h1>
      <div className="tickets">
        <div className="ticket-headings">
          <div>Date</div>
          <div>Item Name</div>
          <div>Status</div>
          <div></div>
        </div>
        {invoices.map((invoice) => (
          <InvoiceItem key={invoice._id} invoice={invoice} />
        ))}
      </div>
    </>
  );
}

export default Invoices;
