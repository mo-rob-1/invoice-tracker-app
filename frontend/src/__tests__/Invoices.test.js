import { render, waitForDomChange } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter as Router } from "react-router-dom";
import Invoices from "../pages/Invoices";

describe("Invoices", () => {
  it("renders the invoices page", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <Invoices />
        </Router>
      </Provider>
    );
    waitForDomChange().then(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it("renders the page title", () => {
    const history = createMemoryHistory();
    const { getByText } = render(
      <Provider store={store} history={history}>
        <Router>
          <Invoices />
        </Router>
      </Provider>
    );
    waitForDomChange().then(() => {
      expect(getByText("Invoices")).toBeInTheDocument();
    });
  });

  it("renders the sub heading titles", () => {
    const history = createMemoryHistory();
    const { getByText } = render(
      <Provider store={store} history={history}>
        <Router>
          <Invoices />
        </Router>
      </Provider>
    );
    waitForDomChange().then(() => {
      expect(getByText("Date")).toBeInTheDocument();
      expect(getByText("Item Name")).toBeInTheDocument();
      expect(getByText("Status")).toBeInTheDocument();
      expect(getByText("Invoice Details")).toBeInTheDocument();
    });
  });
});
