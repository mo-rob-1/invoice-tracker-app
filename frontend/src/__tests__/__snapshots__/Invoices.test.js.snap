// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoices renders the invoices page 1`] = `
<DocumentFragment>
  <a
    class="btn btn-reverse btn-back"
    href="/"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm28.9-143.6L209.4 288H392c13.3 0 24-10.7 24-24v-16c0-13.3-10.7-24-24-24H209.4l75.5-72.4c9.7-9.3 9.9-24.8.4-34.3l-11-10.9c-9.4-9.4-24.6-9.4-33.9 0L107.7 239c-9.4 9.4-9.4 24.6 0 33.9l132.7 132.7c9.4 9.4 24.6 9.4 33.9 0l11-10.9c9.5-9.5 9.3-25-.4-34.3z"
      />
    </svg>
     Back
  </a>
  <h1>
    Invoices
  </h1>
  <div
    class="tickets"
  >
    <div
      class="ticket-headings"
    >
      <div>
        Date
      </div>
      <div>
        Item Name
      </div>
      <div>
        Status
      </div>
      <div>
        Invoice Details
      </div>
    </div>
  </div>
</DocumentFragment>
`;
